includes/rest/class-outbox-controller.php

```diff
@@ -273,4 +271,49 @@ public function get_item_schema() {
 
 		return $this->add_additional_fields_schema( $this->schema );
 	}
+
+	/**
+	 * Overload total items.
+	 *
+	 * The `totalItems` property is used by Mastodon to show the overall
+	 * number of federated posts and comments.
+	 *
+	 * @param array            $response The response array.
+	 * @param \WP_REST_Request $request  The request object.
+	 *
+	 * @return array The modified response array.
+	 */
+	public function overload_total_items( $response, $request ) {
+		$posts = new \WP_Query(
+			array(
+				'post_status' => 'publish',
+				'author'      => $request->get_param( 'user_id' ),
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'  => array(
+					array(
+						'key'     => 'activitypub_status',
+						'compare' => 'EXISTS',
+					),
+				),
+				'fields'      => 'ids',
+				'nopaging'    => true,
```

Comment:
Since we don't do anything with these IDs, I wonder if it would be a faster query if we just looked for 1 post, similar to what we're doing with comments below?